cmd/celestia/p2p.go

```diff
@@ -0,0 +1,601 @@
+package main
+
+import (
+	"github.com/libp2p/go-libp2p/core/metrics"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	ma2 "github.com/multiformats/go-multiaddr"
+	"github.com/spf13/cobra"
+)
+
+type peerInfo struct {
+	ID       string   `json:"id"`
+	PeerAddr []string `json:"peer_addr"`
+}
+
+func init() {
+	p2pCmd.AddCommand(infoCmd,
+		peersCmd,
+		peerInfoCmd,
+		connectCmd,
+		closePeerCmd,
+		connectednessCmd,
+		natStatusCmd,
+		blockPeerCmd,
+		unblockPeerCmd,
+		blockedPeersCmd,
+		protectCmd,
+		unprotectCmd,
+		protectedCmd,
+		bandwidthStatsCmd,
+		peerBandwidthCmd,
+		bandwidthForProtocolCmd,
+		pubsubPeersCmd,
+	)
+}
+
+var p2pCmd = &cobra.Command{
+	Use:   "p2p [command]",
+	Short: "Allows to interact with the P2P Module via JSON-RPC",
+	Args:  cobra.NoArgs,
+}
+
+var infoCmd = &cobra.Command{
+	Use:   "info",
+	Short: "Gets the node's peer info (peer id and multiaddresses)",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		info, err := client.P2P.Info(cmd.Context())
+
+		formatter := func(data interface{}) interface{} {
+			peerAdd, ok := data.(peer.AddrInfo)
+			if !ok {
+				return peerAdd
+			}
+
+			ma := make([]string, len(info.Addrs))
+			for i := range peerAdd.Addrs {
+				ma[i] = peerAdd.Addrs[i].String()
+			}
+
+			return peerInfo{
+				ID:       peerAdd.ID.String(),
+				PeerAddr: ma,
+			}
+		}
+
+		printOutput(info, err, formatter)
+		return nil
+	},
+}
+
+var peersCmd = &cobra.Command{
+	Use:   "peers",
+	Short: "Lists the peers we are connected to",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		result, err := client.P2P.Peers(cmd.Context())
+		peers := make([]string, len(result))
+		for i, peer := range result {
+			peers[i] = peer.String()
+		}
+
+		formatter := func(data interface{}) interface{} {
+			conPeers, ok := data.([]string)
+			if !ok {
+				return conPeers
+			}
+			return struct {
+				Peers []string `json:"peers"`
+			}{
+				Peers: conPeers,
+			}
+		}
+
+		printOutput(peers, err, formatter)
+		return nil
+	},
+}
+
+var peerInfoCmd = &cobra.Command{
+	Use:   "peer-info [param]",
+	Short: "Gets PeerInfo for a given peer",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		client, err := rpcClient(cmd.Context())
+		if err != nil {
+			return err
+		}
+
+		pid, err := peer.Decode(args[0])
+		if err != nil {
+			return err
+		}
+		info, err := client.P2P.PeerInfo(cmd.Context(), pid)
+		formatter := func(data interface{}) interface{} {
+			peerAdd, ok := data.(peer.AddrInfo)
+			if !ok {
```

Comment:
here too, it's everywhere

Reply:
the reason why I did this is because if err != nil, then printOutput  changes the value of the result to error. But the error does not need to be formatted.  I see now that it is confusing, so will change a bit the implementation(actually as this is a nested function, we do not need to check for `ok`)